File: 原創不分類題庫/Contents/d487/d487.cpp
#include<iostream>

using namespace std;

int fac(int);

int main(void)
{
	int n, temp;
	
	while(cin >> n)
	{
		cout << n << "! = ";
		temp = fac(n);
		cout << temp << endl;
	}
	
	return 0;
}

int fac(int num)
{
	if(!num || num == 1)
	{
		cout << "1 = ";
		return 1;
	}
	
	cout << num << " * ";
	return fac(num - 1) * num;
} 

File: 原創不分類題庫/Contents/d487/d487.py
import sys

def fac(num):
    if num == 0 or num == 1:
        print("1 = ", end="")
        return 1
    
    print("%d * " % num, end="")
    return fac(num - 1) * num

for n in sys.stdin:
    n = int(n)

    print("%d! = " % n, end="")
    temp = fac(n)
    print(temp)
